hibernate-core/src/test/java/org/hibernate/jpa/test/query/QueryTest.java

```diff
@@ -704,6 +704,138 @@ public void testJpaPositionalParameters() {
 			em.close();
 		}
 	}
+	
+	/**
+	 * Collection parameters are internally rewritten to named parameters (one named parameter by
+	 * collection item) and HQL parser rejects mixed positional and named parameters queries.
```

Comment:
The description of the test is not very accurate. Reading it, I thought it would be an error case but it is not.